src/shared-ui/src/components/icons/IconBase.vue

```diff
@@ -0,0 +1,73 @@
+<!--
+   The MIT License
+
+   Copyright (c) 2019- Nordic Institute for Interoperability Solutions (NIIS)
+   Copyright (c) 2018 Estonian Information System Authority (RIA),
+   Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
+   Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+   THE SOFTWARE.
+ -->
+<template>
+  <svg
+    xmlns="http://www.w3.org/2000/svg"
+    :width="width"
+    :height="height"
+    viewBox="0 0 26 26"
+    :aria-labelledby="iconName"
+    role="presentation"
+    :fill="color"
+  >
+    <title :id="iconName" lang="en">{{ iconName }} icon</title>
+    <slot :fillColor="color" />
+  </svg>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+export default Vue.extend({
+  name: 'XrdIconBase',
```

Comment:
I get the idea of having the `XrdIconBase` component but IMO the usage is not too convenient at the moment:
```
<xrd-icon-base color="red">
  <XrdIconErrorNotification />
</xrd-icon-base>
```

What would it take to get it syntactically improved to something like this?
```
<XrdIconErrorNotification color="red" />
```
Perhaps the base component could be baked into the icon components

Reply:
I could be wrong, but it's not possible to inherit a part of the template. So I don't know how to share the wrapping part. It's possible to inherit methods etc. via mixins, but that would not help in this case. Also if there is some sharing a solution it would add more code and "moving parts" into every icon file. Currently they are super simple.

This solution is done following the example in Vue "official" cookbook. https://vuejs.org/v2/cookbook/editable-svg-icons.html